community/kernel/src/main/java/org/neo4j/storageengine/api/schema/LabelScanReader.java

```diff
@@ -33,6 +33,15 @@
      */
     PrimitiveLongResourceIterator nodesWithLabel( int labelId );
 
+    /**
+     * Returns the index progressor over the label id
+     *
+     * @param client the client to communicate with
+     * @param labelId label token id
+     * @return index progressor with the given {@code labelId}
+     */
+    IndexProgressor nodesWithLabelIndex( IndexProgressor.NodeLabelClient client, int labelId );
```

Comment:
Got some comments about this guy.

1. This is the same pattern that we use for `IndexReader`. In that case, the full responsibility for connecting progressor and client lies with implementation and the progressor is never visible from from the outside. Here we instead return the progressor and leave the responsibility of initializing the client with the progressor to the caller. I would prefer that we use the same usage pattern in both cases. Unless there is some reason for the different approaches that I don't see.

2. I think adding "index" to this method name adds more confusion than it removes. I would prefer that we use the same method name as above , `nodesWithLabel`.